modules/src/ics24_host/identifier.rs

```diff
@@ -179,6 +179,26 @@ impl ClientId {
     pub fn as_bytes(&self) -> &[u8] {
         &self.0.as_bytes()
     }
+
+    /// Returns the suffix counter of this client.
+    /// Returns `None` if the client identifier is malformed
+    /// and the suffix could not be parsed.
+    /// ```rust
+    /// # use ibc::ics24_host::identifier::ClientId;
+    /// # use ibc::ics02_client::client_type::ClientType;
+    /// # use std::str::FromStr;
+    /// let cl_a = ClientId::new(ClientType::Mock, 670).unwrap();
+    /// assert_eq!(cl_a.suffix(), Some(670));
+    /// let cl_b = ClientId::from_str("nosuffix_client").unwrap();
+    /// assert_eq!(cl_b.suffix(), None);
+    /// ```
+    pub fn suffix(&self) -> Option<u64> {
+        self.as_str()
+            .split('-')
+            .last()
+            .map(|e| e.parse::<u64>().ok())
+            .flatten()
+    }
```

Comment:
Maybe move this to `cosmos.rs` as query helper. Not all chains may have the same generation pattern as cosmos.

Reply:
Similar to the `ClientId::suffix()` method introduce in this PR, the method `ClientId::new()` also assumes the generation pattern with prefix/suffix for client ids, separated by a dash.

https://github.com/informalsystems/ibc-rs/blob/7c0418e3f6adf270a2a91f19b39a0141234c7545/modules/src/ics24_host/identifier.rs#L144-L159

I agree that this pattern is probably Cosmos-specific. But not sure we should separate this functionality in this PR. If you think this is the right time, it would be an easy refactor and I'll do it.

Reply:
The `new()` method should only be used by the on-chain handlers and it is ok to have it there as they are for a specific chain implementation, but not necessarily cosmos; ics24 probably not the best place for that, maybe the corresponding keeper, e.g. `ClientKeeper` for client IDs, `ConnectionKeeper` for connection IDs, etc. 
`suffix()` on the other hand is only for `CosmosSdk` chains.

Reply:
> The new() method should only be used by the on-chain handlers and it is ok to have it there as that is for a specific chain implementation, but not necessarily cosmos

That's not how I interpreted it. The `new` method is `pub` so there's no reason why the relayer should not be using it. But I think I now understand your point. Thanks Anca for the explanation!

I refactored the problematic `ClientId::suffix()` into `cosmos.rs`. ad9ddf3e